sympy/calculus/util.py

```diff
@@ -253,6 +254,192 @@ def elm_domain(expr, intrvl):
+    from sympy.solvers.decompogen import decompogen
+
+    orig_f = f
+    f = simplify(orig_f)
+    period = None
+
+    if isinstance(f, TrigonometricFunction):
+        arg = f.args[0]
+        try:
+            period = f.period(symbol)
+        except NotImplementedError:
+            pass
+
+    if f.is_Pow:
+        base, expo = f.args
+        base_has_sym = base.has(symbol)
+        expo_has_sym = expo.has(symbol)
+
+        try:
+            if base_has_sym and not expo_has_sym:
+                period = periodicity(base, symbol)
+
+            elif expo_has_sym and not base_has_sym:
+                period = periodicity(expo, symbol)
+
+            else:
+                period = _periodicity(f.args, symbol)
+
+        except NotImplementedError:
+            pass
+
+    if f.is_Mul:
+        _, g = f.as_independent(symbol, as_Add=False)
+        if isinstance(g, TrigonometricFunction):
+            period = periodicity(g, symbol)
+
+        else:
+            period = _periodicity(g.args, symbol)
+
+    if f.is_Add:
+        k, g = f.as_independent(symbol)
+        if k is not S.Zero:
+            return periodicity(g, symbol)
+
+        period = _periodicity(g.args, symbol)
+
+    if period is None:
+        from sympy.solvers.decompogen import compogen
+        g_s = decompogen(f, symbol)
+        num_of_gs = len(g_s)
+        if num_of_gs > 1:
+            for index, g in enumerate(reversed(g_s)):
+                try:
+                    start_index = num_of_gs - 1 - index
+                    g = compogen(g_s[start_index:], symbol)
+                    if g != f:
+                        period = periodicity(g, symbol)
+
+                except (ValueError, NotImplementedError):
+                    continue
+                else:
+                    break
+
+    if period is not None:
+        if check:
+            if orig_f.subs(symbol, symbol + period) == orig_f:
+                return period
+
+            else:
+                raise NotImplementedError(filldedent('''
+                    The period of the given function cannot be verified.
+                    Set check=False to obtain the value.'''))
+
+        return period
+
+    else:
+        raise NotImplementedError(filldedent('''Period of the given function
+            cannot be determined using this method. The function might be
+            non-periodic.'''))
+
+
+def _periodicity(args, symbol):
+    """Helper for periodicity to find the period of a list of simpler
+    functions. It uses the `nlcm` method to find the least common period of
+    all the functions.
+    """
+    periods = []
+    for f in args:
+        try:
+            period = periodicity(f, symbol)
+            periods.append(period)
+        except NotImplementedError:
+            return None
+
+    if len(periods) > 1:
+        return nlcm(periods)
+
+    elif not periods:
+        return None
+
+
+def nlcm(numbers):
```

Comment:
> what is the output for `nlcm([2*E, 3*E, E, E/2])`?

`None` is being returned. 
I will try to generalize this implementation.
In Meeting# 10, @aktech had suggested that an implementation specific to our cause would do for now.
Hence, I had developed this method with the thought of extending it in the future.